app/utils/modify-shared-ways.js

```diff
@@ -65,10 +78,12 @@ export default function modifySharedWays (sharedWays, node, coordinates, action)
             node.properties.ways = { ...node.properties.ways }
             node.properties.ways[newWay.properties.id] = indexOfNearestPoint + 1
             newWay.properties.ndrefs.splice(indexOfNearestPoint + 1, 0, node.properties.id)
+
+            index = indexOfNearestPoint + 1
           }
           modifiedSharedWays.push(newWay)
         })
       }
   }
-  return modifiedSharedWays
+  return { modifiedSharedWays, index }
```

Comment:
after trying a few things it seemed like `indexOfNearestPoint + 1` was all I needed so decided to return an object that could be destructured that includes both the index and the `modifiedSharedWays`.

it might be better if `index` had a more descriptive name.